features/lorawan/LoRaWANStack.cpp

```diff
@@ -0,0 +1,2126 @@
+    lora_mcps_confirm.nb_retries = mcps_confirm->NbRetries;
+    lora_mcps_confirm.datarate = mcps_confirm->Datarate;
+    lora_mcps_confirm.tx_power = mcps_confirm->TxPower;
+    lora_mcps_confirm.uplink_counter = mcps_confirm->UpLinkCounter;
+    lora_mcps_confirm.uplink_frequency = mcps_confirm->UpLinkFrequency;
+
+    // Interprets from Mcps_t to lora_mac_mcps_t
+    mcps_confirm->McpsRequest = interpret_mcps_confirm_type(lora_mcps_confirm.mcps_request);
+    // Interprets from LoRaMacEventInfoStatus_t to lora_mac_event_info_status_t
+    lora_mcps_confirm.status = interpret_event_info_type(mcps_confirm->Status);
+    lora_mcps_confirm.tx_time_on_air = mcps_confirm->TxTimeOnAir;
+
+    mcps_confirm_handler(&lora_mcps_confirm);
+}
+
+/** Wrapper function to MCPS-Indication event function
+ *
+ * \param mcps_indication   Pointer to the indication structure,
+ *                          containing indication attributes.
+ */
+void LoRaWANStack::mcps_indication(McpsIndication_t *mcps_indication)
+{
+    lora_mac_mcps_indication_t lora_mcps_indication;
+
+    lora_mcps_indication.ack_received = mcps_indication->AckReceived;
+    memcpy(lora_mcps_indication.buffer,  mcps_indication->Buffer, mcps_indication->BufferSize);
+    lora_mcps_indication.buffer_size = mcps_indication->BufferSize;
+    lora_mcps_indication.downlink_counter = mcps_indication->DownLinkCounter;
+    lora_mcps_indication.frame_pending = mcps_indication->FramePending;
+    lora_mcps_indication.mcps_indication = (lora_mac_mcps_t)mcps_indication->McpsIndication;
+    lora_mcps_indication.multicast = mcps_indication->Multicast;
+    lora_mcps_indication.port = mcps_indication->Port;
+    lora_mcps_indication.rssi = mcps_indication->Rssi;
+    lora_mcps_indication.rx_data = mcps_indication->RxData;
+    lora_mcps_indication.rx_datarate = mcps_indication->RxDatarate;
+    lora_mcps_indication.rx_slot = mcps_indication->RxSlot;
+    lora_mcps_indication.snr = mcps_indication->Snr;
+    lora_mcps_indication.status = (lora_mac_event_info_status_t)mcps_indication->Status;
+
+    mcps_indication_handler(&lora_mcps_indication);
+}
+
+/** Wrapper function to MLME-Confirm event function
+ *
+ * \param mlme_confirm      Pointer to the confirm structure,
+ *                          containing confirm attributes.
+ */
+void LoRaWANStack::mlme_confirm(MlmeConfirm_t *mlme_confirm)
+{
+    lora_mac_mlme_confirm_t lora_mlme_confirm;
+
+    lora_mlme_confirm.demod_margin = mlme_confirm->DemodMargin;
+    lora_mlme_confirm.mlme_request = (lora_mac_mlme_t)mlme_confirm->MlmeRequest;
+    lora_mlme_confirm.nb_gateways = mlme_confirm->NbGateways;
+    lora_mlme_confirm.nb_retries = mlme_confirm->NbRetries;
+    lora_mlme_confirm.status = (lora_mac_event_info_status_t)mlme_confirm->Status;
+    lora_mlme_confirm.tx_time_on_air = mlme_confirm->TxTimeOnAir;
+
+    mlme_confirm_handler(&lora_mlme_confirm);
+}
+
+void LoRaWANStack::set_lora_event_cb(Callback<void(lora_events_t)> event_cb)
```

Comment:
Is this `attach()` method ?

Reply:
It is used to post events to user. User provided callback function.

Reply:
In our API we use `attach()`, this is not the case here?

Reply:
Actual API visible to user is lora_event_callback() defined in LoRaWANInterface class. 
This name was suggested by Jan and Sam. I am not keen on any name :), if majority wants it to be attach, we can name it attach. Let's wait what others say.